content/3_Silver/Intro_Ordered.mdx

```diff
@@ -93,20 +93,62 @@ export const problems = {
 
 In ordered sets and maps, accessing or removing the next key higher or lower than some input `k` is also supported.
 
-<LanguageSection>
-
-<CPPSection>
 
 ## Using Iterators
 
 In Bronze, we avoided discussion of any set operations involving iterators.
 
+<LanguageSection>
+
+<CPPSection>
+
 <Resources>
 	<Resource source="CPH" title="4.4 - Set Iterators" starred></Resource>
 </Resources>
 
 </CPPSection>
 
+<JavaSection>
+
+In java, iterators are helpful in looping through sets.
+
+Iterators used on **HashSet** would have the elements in a random order:
+```java
+Set<Integer> set = new HashSet<Integer>();
+set.add(1); set.add(3); set.add(0); set.add(-2);
+Iterator it = set.iterator();
+while(it.hasNext()){
+   Integer i = (Integer)it.next();
+   System.out.print(i + " "); // returns 0 1 -2 3
+}
+```
+But in **TreeSet** the elements are in a sorted order:
+```java
+Set<Integer> set = new TreeSet<Integer>();
+set.add(1); set.add(3); set.add(0); set.add(-2);
+Iterator it = set.iterator();
+while(it.hasNext()){
+   Integer i = (Integer)it.next();
+   System.out.print(i + " "); // returns -2 0 1 3
+}
+```
+
+Instead of creating an iterator and looping through it, Java provides a for-each loop which creates a hidden iterator and loops through it automatically:
```

Comment:
```suggestion
Instead of creating an iterator and looping with it like in C++, Java provides a for-each loop which creates a hidden iterator and loops with it automatically:
```